### PR TITLE
Transaction modal

### PR DESCRIPTION
**Pull request by jakai7**
Added a simple modal for transactions, not much styling yet just the basic functionality.

### CODE REVIEW

**Comment by musnit** [comment]
can we avoid using material-ui?

File: src/AppETH.tsx [hunk 1]
```diff
@@ -76,7 +76,7 @@ function AppETH({ net }: Props): React.ReactElement<Props> {
         {/* SS58 Address Input */}
         <Grid item xs={10}>
           <FormControl>
-            <SelectedAccount address={net.polkadotAddress} />
+            <PolkadotAvatar address={net.polkadotAddress} />
```
Comment (musnit) [inline src/AppETH.tsx:79]: should this be PolkadotAccount rather?

File: src/components/TransactionMenu/index.tsx [hunk 2]
```diff
@@ -50,29 +88,81 @@ export default function TransactionMenu({
...
+                  ) : (
+                    <span style={{ color: 'red' }}>{t.confirmations}</span>
+                  )}
+                </TableCell>
+                <TableCell align="left">
+                  <small>{t.amount}</small>
+                </TableCell>
+                <TableCell align="center">
+                  <button onClick={handleDelete} data-hash={t.hash}>
+                    Delete
```
Comment (musnit) [inline src/components/TransactionMenu/index.tsx:155]: whats the purpose of this delete button?

File: src/components/TransactionMenu/index.tsx [hunk 3]
```diff
@@ -92,10 +182,18 @@ export default function TransactionMenu({
         open={Boolean(anchorEl)}
         onClose={handleClose}
       >
-        <ClearTransactionsBtn />
         <ZeroTransactions />
         {net.transactions.map((t) => TransactionMenuItem(t))}
+        <AllTransactionsBtn />
+        <ClearTransactionsBtn />
```
Comment (musnit) [inline src/components/TransactionMenu/index.tsx:188]: whats ClearTransactions for?

File: src/net/index.ts [hunk 4]
```diff
@@ -52,14 +42,23 @@ export default class Net {
...
+
   // Empty transactions list
   public emptyTransactions(): void {
     this.transactions = [];
   }
 
   // returns number of pending (confirming) transactions
   public pendingTransactions(): number {
-    return this.transactions.filter((t) => t.status === 'confirming').length;
+    return this.transactions.filter((t) => t.confirmations < 12).length;
```
Comment (musnit) [inline src/net/index.ts:61]: can we make the number of confirmations be set in config as a constant?

**Comment by jakai7** [comment]
> can we avoid using material-ui?
 
Sure, I can remove it. Any suggestions on an alternative?

File: src/AppETH.tsx (hunk 1 above)
Comment (jakai7) [reply src/AppETH.tsx]: Yes that is much better. I changed it.

File: src/components/TransactionMenu/index.tsx (hunk 2 above)
Comment (jakai7) [reply src/components/TransactionMenu/index.tsx:155]: It removes a transaction from the array of all submitted transactions, it is not required just added it as an ux feature.

File: src/components/TransactionMenu/index.tsx (hunk 3 above)
Comment (jakai7) [reply src/components/TransactionMenu/index.tsx:188]: It clears the array of all submitted transactions, similar to the delete button it is not required just added it as an ux feature.

File: src/components/TransactionMenu/index.tsx (hunk 2 above)
Comment (musnit) [reply src/components/TransactionMenu/index.tsx:155]: hm i'm not sure we should have any clear/delete functionality at all - don't think it necessarily provides user value, and could also be confusing in terms of what it means after refresh/coming back to app.

File: src/components/TransactionMenu/index.tsx (hunk 3 above)
Comment (musnit) [reply src/components/TransactionMenu/index.tsx:188]: hm i'm not sure we should have any clear/delete functionality at all - don't think it necessarily provides user value, and could also be confusing in terms of what it means after refresh/coming back to app.

**Comment by musnit** [comment]
> > can we avoid using material-ui?
> 
> Sure, I can remove it. Any suggestions on an alternative?

yeh just plain html/css

File: src/net/index.ts (hunk 4 above)
Comment (jakai7) [reply src/net/index.ts]: Updated.

File: src/components/TransactionMenu/index.tsx (hunk 2 above)
Comment (jakai7) [reply src/components/TransactionMenu/index.tsx]: Makes sense. I removed the methods

**Pull request merged**